test/device_connector_test.dart

```diff
@@ -335,9 +261,30 @@ void main() {
   });
 }
 
-class _PluginControllerMock extends Mock implements PluginController {}
+class _DeviceConnectionControllerStub implements DeviceConnectionController {
```

Comment:
It would be better to defer the tedious work of writing stubs to a code-generator, wouldn't it? And why to develop our own, if mockito is doing this for everyone already?

What are your arguments against mockito with code-generation, actually?

Reply:
I answered it in the [issue](https://github.com/PhilipsHue/flutter_reactive_ble/issues/174) .

Reply:
https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md has some alternatives to entirely manually written stubs. With version 4.1.2, both code generation and the manual calls to `super.noSuchMethod(Invocation.method(#start, [args...], return value)` seem to work (possibly with an additional cast).